presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp

```diff
@@ -1638,7 +1645,15 @@ core::PlanNodePtr VeloxQueryPlanConverterBase::toVeloxQueryPlan(
   }
   if (auto aggregation =
           std::dynamic_pointer_cast<const protocol::AggregationNode>(node)) {
-    return toVeloxQueryPlan(aggregation, tableWriteInfo, taskId);
+    bool isStreamingAggAllowed = true;
+    if (auto joinNode = std::dynamic_pointer_cast<protocol::JoinNode>(
```

Comment:
what if nested loop join is not a direct source, but is in the plan tree (e.g. there's a project node in between)?

Reply:
@rschlussel : We saw this property set by the optimizer in aggregation over nested loop join via StreamPropertyDerivations https://github.com/prestodb/presto/pull/23315. But you are right that the ideal fix should look for all nodes that can carry forward projections of the grouping keys.

Reply:
My question is what if you have like so


aggregation
   |
filter or identity project (any node that propagates the properties of its input)
  |
nested loop join
 |
something sorted

the nested loop join will mess up the sortedness, but we won't catch it with this code because it's not a direct source of the aggregation node.  so we will still have a correctness bug.

Reply:
@rschlussel : Yes, the ideal fix should check for any nested loop join between aggregation and the TableScan it is originating from. I believe @karteekmurthys  is going to change this code.

Reply:
@rschlussel I am trying to see if I can do a recursive search for a nestedloopjoin node and  if present skip streaming.